### PR TITLE
Getting-Started on Contributing.

### PR DESCRIPTION
**Pull request by Mikcl**
Addresses the issues discussed in https://github.com/philipturner/molecular-renderer/pull/8#issuecomment-1872286369

I made a few changes: 

1.a) renamed `bypass_dependencies` to `vendor`. 
- condisered `dylibs` and `.dylibs`, but settled on `vendor`. Open to suggestions/changing it?

1.b) Rather than a `../bypass_dependencies`, I have added a git ignored `./vendor` directory _inside_ the repo which should contain the `dylibs`. The motivation for this is that the depenencies for this project are contained within repository. 

2)a) updated the build rules, to look at the `vendor` directory.
2)b) changed the `DEVELOPMENT_TEAM` to none, since contributors are not part of the `DEVELOPMENT_TEAM`, are there any concerns? 
2)c) Some auto-generated changes, since the embeded libraries selected in the GUI come from the vendor directory now. (Note this PR is breaking change if you do not have a `./vendor` directory). 

3) added instructions to get started in the README, In now takes about 1 minute to get started (provided you have xcode, swift and conda preinstalled) 

---

Im not sure if any other steps are missing?

### CODE REVIEW

File: MolecularRenderer.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -104,16 +106,25 @@
...
 			dstSubfolderSpec = 10;
 			files = (
-				98A4A5792A81A65E00A74F87 /* libfreetype.6.dylib in Embed Libraries */,
-				98C48C5A2A48F7510047C078 /* libOpenMM.8.0.dylib in Embed Libraries */,
-				98B3E4F82A3F700400B02349 /* libc++.1.dylib in Embed Libraries */,
+				3C08BA492B3F780200949C9B /* libfreetype.6.dylib in Embed Libraries */,
+				3C08BA4A2B3F780200949C9B /* libOpenMM.8.0.dylib in Embed Libraries */,
+				3C08BA4B2B3F780200949C9B /* libc++.1.dylib in Embed Libraries */,
+				3C08BA4C2B3F780200949C9B /* libpng16.16.dylib in Embed Libraries */,
+				3C08BA4D2B3F780200949C9B /* libz.1.dylib in Embed Libraries */,
```
Comment (Mikcl) [inline MolecularRenderer.xcodeproj/project.pbxproj:113]: These two are infact dependencies, so added them via the GUI: 

```
otool -L ./vendor/libfreetype.6.dylib
./vendor/libfreetype.6.dylib:
	@rpath/libfreetype.6.dylib (compatibility version 25.0.0, current version 25.3.0)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	@rpath/libpng16.16.dylib (compatibility version 56.0.0, current version 56.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.0.0)
```

**Comment by philipturner** [comment]
I don't use a Conda env, I just have a `~/Users/philipturner/miniforge3` folder where all the dylibs go. I call `conda install` straight from the command-line. No `requirements.txt` needed.

Screenshot 2023-12-29 at 10 42 32 PM

This is great work, but I'll have to defer further review until later. I think if a third person ever comes around with interest in this project, fixing up the documentation will be necessary. In the meantime, if it works for yourself, we're good to start moving on to the next stage.

Perhaps open a GitHub issue or some other method of communication? There's a lot of ground to cover before you can start designing nanomachines properly. I can also show some previous demos for you to visualize in the renderer. Whatever you are interested in.

**Comment by Mikcl** [comment]
> I'll have to defer further review until later. I think if a third person ever comes around with interest in this project, fixing up the documentation will be necessary.

Very well :) will put this into draft for now.

**Comment by philipturner** [comment]
I removed half of the external dependencies. This also presents a potential place for future contributions. Finding whether there's a C library that can encode MP4 videos. There might be, but I didn't have the time to learn it. I had basically zero free time back then. Now I have more, thanks to changing my plan of study to MNT. Which has provided freedom to work more on getting the right software (mostly, the compiler).

https://github.com/philipturner/molecular-renderer/blob/main/Sources/HardwareCatalog/Other/Serialization/README.md

[permalink](https://github.com/philipturner/molecular-renderer/blob/4bd147994d8aba71b9c6880f7db3696b6eecf5f2/Sources/HardwareCatalog/Other/Serialization/README.md)

**Comment by philipturner** [comment]
Closing this issue, as I am overhauling the entire repository to be more encapsulated. Decoupling the simulators from the renderer itself. I will have decent documentation when I am done.

**Pull request closed without merging**